Stratis.Bitcoin/Features/BlockStore/LoopSteps/DownloadBlockStep.cs

```diff
@@ -48,31 +44,26 @@ internal DownloadBlockStep(BlockStoreLoop blockStoreLoop)
                     var breakExecution = await ShouldBreakExecution(inputChainedBlock, nextChainedBlock);
                     if (breakExecution)
                     {
-                        if (!chainedBlockStackToDownload.Any())
+                        if (!downloadStack.Any())
                             return BlockStoreLoopStepResult.Break();
 
                         canDownload = false;
                     }
                     else
                     {
                         this.BlockStoreLoop.BlockPuller.AskBlock(nextChainedBlock);
-                        chainedBlockStackToDownload.Enqueue(nextChainedBlock);
+                        downloadStack.Enqueue(nextChainedBlock);
 
-                        if (chainedBlockStackToDownload.Count == this.BlockStoreLoop.BatchDownloadSize)
+                        if (downloadStack.Count == this.BlockStoreLoop.BatchDownloadSize)
                             canDownload = false;
                     }
                 }
 
-                var downloadResult = await DownloadBlocks(chainedBlockStackToDownload, cancellationToken);
+                var downloadResult = await DownloadBlocks(insertBlockSize, stallCount, store, downloadStack, cancellationToken);
```

Comment:
This wont work unless you use (ref insertBlockSize,..)
How ever I would prefer to create a mini context